addons/binding/org.openhab.binding.dlinkupnpcamera/src/main/java/org/openhab/binding/dlinkupnpcamera/internal/DlinkUpnpCameraHandlerFactory.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dlinkupnpcamera.internal;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceRegistry;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.io.transport.upnp.UpnpIOService;
+import org.openhab.binding.dlinkupnpcamera.DlinkUpnpCameraBindingConstants;
+import org.openhab.binding.dlinkupnpcamera.config.DlinkUpnpCameraConfiguration;
+import org.openhab.binding.dlinkupnpcamera.handler.DlinkUpnpCameraHandler;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link upnpcameraHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Yacine Ndiaye
+ * @author Antoine Blanc
+ * @author Christopher Law
+ */
+public class DlinkUpnpCameraHandlerFactory extends BaseThingHandlerFactory {
+    private Logger logger = LoggerFactory.getLogger(DlinkUpnpCameraHandlerFactory.class);
+
+    private UpnpIOService upnpIOService;
+    private DiscoveryServiceRegistry discoveryServiceRegistry;
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+
+        return DlinkUpnpCameraBindingConstants.SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected ThingHandler createHandler(Thing thing) {
+        logger.debug("ThingHandler createHandler");
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (thingTypeUID.equals(DlinkUpnpCameraBindingConstants.CAMERA_THING_TYPE_UID)) {
+            return new DlinkUpnpCameraHandler(thing, upnpIOService, discoveryServiceRegistry);
+        }
+
+        return null;
+    }
+
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        logger.debug("activate upnpcameraHandler");
```

Comment:
Do you really need this?

Apart from that: this is simply wrong. It's the factory, not the handler.